test_plugins/UIPlugin.py

```diff
@@ -191,6 +214,156 @@ def prefab_button_pressed_callback(button):
+        slider.register_released_callback(slider_released_callback)
+
+        Logs.debug("Added slider")
+
+        ln_textInput.padding_type = nanome.ui.LayoutNode.PaddingTypes.ratio
+        ln_textInput.padding = (0.01, 0.01, 0.01, 0.01)
+        ln_textInput.forward_dist = .001
+
+        textInput = nanome.ui.TextInput()
+        textInput.max_length = 30
+        textInput.register_changed_callback(text_changed_callback)
+        textInput.register_submitted_callback(text_submitted_callback)
+
+        Logs.debug("Added text input")
+
+        prefab = nanome.ui.LayoutNode()
+        prefab.layout_orientation = nanome.ui.LayoutNode.LayoutTypes.vertical
+        child1 = nanome.ui.LayoutNode()
+        child1.sizing_type = nanome.ui.LayoutNode.SizingTypes.ratio
+        child1.sizing_value = .3
+        child1.name = "label"
+        child1.forward_dist = .01
+        child2 = nanome.ui.LayoutNode()
+        child2.name = "button"
+        child2.forward_dist =.01
+        prefab.add_child(child1)
+        prefab.add_child(child2)
+        prefabLabel = nanome.ui.Label()
+        prefabLabel.text_value = "Molecule Label"
+        prefabButton = nanome.ui.Button()
+        prefabButton.text.active = True
+        prefabButton.set_all_text("Molecule Button")
+        prefabButton.register_pressed_callback(prefab_button_pressed_callback)
+        child1.set_content(prefabLabel)
+        child2.set_content(prefabButton)
+
+        ln_loading_bar = nanome.ui.LayoutNode(name="LoadingBar")
+        ln_loading_bar.forward_dist = .03
+        self.loadingBar = ln_loading_bar.add_new_loading_bar()
+
+        content.add_child(ln_contentBase)
+        ln_contentBase.add_child(ln_label)
+        ln_contentBase.add_child(ln_button)
+        ln_contentBase.add_child(ln_slider)
+        ln_contentBase.add_child(ln_textInput)
+        ln_contentBase.add_child(ln_loading_bar)
+        ln_label.set_content(label)
+        ln_button.set_content(button)
+        ln_slider.set_content(slider)
+        ln_textInput.set_content(textInput)
+        return content
+
+    def create_tab_buttons(self):
+        LN = nanome.ui.LayoutNode
+        ln = LN()
+        ln.layout_orientation = nanome.util.enums.LayoutTypes.horizontal
+        ln._sizing_type = nanome.util.enums.SizingTypes.fixed
+        ln._sizing_value = .1
+
+        def tab1_callback(button):
+            self.tab_button1.selected  = True
+            self.tab_button2.selected  = False
+            self.tab1.enabled = True
+            self.tab2.enabled = False
+
+            self.update_node(self.tabs)
+            self.update_content(self.tab_button1)
+            self.update_content(self.tab_button2)
+
+        def tab2_callback(button):
+            self.tab_button2.selected  = True
+            self.tab_button1.selected  = False
+            self.tab2.enabled = True
+            self.tab1.enabled = False
+
+            n_node = self.tabs.create_child_node("AA")
```

Comment:
this is doing weird stuff with the layout, not sure if intentional?

Reply:
Oops, that shouldn't be there, nice catch